indigo_api/models/tasks.py

```diff
@@ -339,15 +339,28 @@ def post_task_transition(sender, instance, name, **kwargs):
     """
     if name in instance.VERBS:
         user = kwargs['method_args'][0]
+        comment = kwargs['method_kwargs'].get('comment', None)
+
         # ensure the task object changes are in the DB, since action signals
         # load related data objects from the db
         instance.save()
 
-        if name == 'unsubmit':
+        if name == 'unsubmit' and not comment:
```

Comment:
rather than a special-case when we have a comment, why not follow the current path and always include comment, even if it's None. Less complex that way.